src/Microsoft.DotNet.XHarness.CLI/Commands/WASI/Engine/WasiTestCommand.cs

```diff
@@ -65,8 +65,35 @@ protected override async Task<ExitCode> InvokeInternal(ILogger logger)
             logger.LogInformation($"Using wasm engine {Arguments.Engine.Value} from path {engineBinary}");
             await PrintVersionAsync(Arguments.Engine.Value.Value, engineBinary);
 
+            ServerURLs? serverURLs = null;
+            if (Arguments.IsWebServerEnabled)
+            {
+                serverURLs = await WebServer.Start(
+                    Arguments,
+                    logger,
+                    cts.Token);
+            }
+
             var engineArgs = new List<string>();
             engineArgs.AddRange(Arguments.EngineArgs.Value);
+
+            if (Arguments.IsWebServerEnabled)
+            {
+                foreach (var envVariable in Arguments.WebServerHttpEnvironmentVariables.Value)
+                {
+                    engineArgs.Add($"--env");
+                    engineArgs.Add($"{envVariable}={serverURLs!.Http}");
```

Comment:
nit: Just a question:
Could these be: `engineArgs.Add($"--setenv={envVariable}={serverURLs!.Http}");` ?

Reply:
nope, they need to be separate